Formula/haskell-stack.rb

```diff
@@ -26,7 +27,16 @@ def install
     # Reported 25 May 2016: https://github.com/commercialhaskell/stack/issues/2192
     (buildpath/"cabal.config").write("allow-newer: base,transformers\n")
 
-    install_cabal_package
+    cabal_sandbox do
+      cabal_install
+
+      # Let `stack` handle its own parallelization
```

Comment:
Note that the breakage occurred *without* setting `-j` at all, but *with* setting `STACK_ROOT`.